src/main/java/uk/ac/ebi/biosamples/jsonschemastore/ena/ChecklistConverterService.java

```diff
@@ -33,25 +34,77 @@ public ChecklistConverterService(SchemaService schemaService, SchemaObjectPopula
         this.populator = populator;
     }
 
-    public String getChecklist(String checklistId) {
+    private static String getTypedTemplate(Field field) {
+        String fieldTypeTemplate;
+        if (field.getFieldType().getTextChoiceField() != null) {
+            fieldTypeTemplate = SchemaTemplateGenerator.getEnumTemplate(getEnumValueList(field.getFieldType().getTextChoiceField()));
+        } else if (field.getFieldType().getTextField() != null) {
+            String regex = field.getFieldType().getTextField().getRegex() != null ? field.getFieldType().getTextField().getRegex() : "";
+            fieldTypeTemplate = SchemaTemplateGenerator.getStringTemplate(regex, 0, 0, "");
+        } else {
+            fieldTypeTemplate = SchemaTemplateGenerator.getStringTemplate("", 0, 0, "");
+        }
+
+        return fieldTypeTemplate;
+    }
+
+    private static List<String> getEnumValueList(TextChoiceField enumField) {
+        List<String> values = new ArrayList<>();
+        for (TextValue textValue : enumField.getTextValue()) {
+            values.add(textValue.getValue());
+            if (textValue.getSynonyms() != null) {
+                values.addAll(textValue.getSynonyms());
+            }
+        }
+        return values;
+    }
+
+    private static Property mapEnaFieldToProperty(Field field) {
+        return new Property(field.getName(), field.getSynonyms(), field.getDescription(), getTypedTemplate(field),
+            field.getUnits(), getCardinality(field.getMandatory()));
```

Comment:
Just a vocabulary comment. Cardinality is more appropriate to the "single/multiple" attribute of a field's inclusion in a checklist, i.e. the number of values an attribute can hold for a single entity. 

For required/optional/recommended, my best suggestion is inclusionRequirement or inclusionConstraint.